### PR TITLE
Link the background colors and the ui font colors in the relevant css.

### PR DESCRIPTION
**Pull request by HaudinFlorence**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/13152
This PR replaces #13173
## Code changes

Colors in the interface may change slightly.

### CODE REVIEW

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by fcollonval** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
The PR has just been rebased with master and should be ready for review.

**Comment by fcollonval** [comment]
bot please update snapshots

**Comment by jtpio** [comment]
Looking at the snapshot diff some changes don't look desired?

image

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
> Looking at the snapshot diff some changes might not be desired?
> 
> image

Problematic changes that were reducing the contrast may have been reverted in commit 
dca202b

**Comment by HaudinFlorence** [comment]
bot please update snapshots

**Comment by HaudinFlorence** [comment]
The PR has been rebased. As a remark comparing the snapshots, it seems that the notebook icons don't seeam to have the right orange color (maybe a reminiscence of PR #13468). Not sure also that the switch label has been reset to its previous initial color.

File: packages/rendermime/style/base.css [hunk 1]
```diff
@@ -576,6 +576,8 @@
   margin: 1em 2em;
   padding: 0 1em;
   border-left: 5px solid var(--jp-border-color2);
+
+  /*color: var(--jp-ui-font-color0);*/
```
Comment (fcollonval) [inline packages/rendermime/style/base.css:580]: Dead code

```suggestion
```

File: packages/settingeditor/style/base.css [hunk 2]
```diff
@@ -458,7 +460,9 @@ ul.jp-PluginList li.jp-mod-selected span.jp-PluginList-icon.jp-FileIcon {
   text-overflow: ellipsis;
   overflow-x: hidden;
   white-space: nowrap;
-  color: var(--jp-content-font-color1);
+
+  /*color: var(--jp-content-font-color1);*/
```
Comment (fcollonval) [inline packages/settingeditor/style/base.css:464]: Dead code

```suggestion
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @HaudinFlorence could you please remove some of the dead code.

About the notebook icon color, indeed #13279 introduced some regression. Could you correct them in a separate PR.

What I saw:
- Light theme:
  - The notebook icon and the switch true position should be md-orange-900 not md-orange-700
- Dark theme:
  - Jupyter, notebook, json and console icons are not proper. There is a shift to lighter color for the brand and the warning scales. So could you also correct that please.
- We forget to add fallback value for the switch color CSS properties `--jp-switch-color`, `--jp-switch-true-position-color`

**Comment by HaudinFlorence** [comment]
> Thanks @HaudinFlorence could you please remove some of the dead code.
> 
> About the notebook icon color, indeed #13279 introduced some regression. Could you correct them in a separate PR.
> 
> What I saw:
> 
> * Light theme:
>   
>   * The notebook icon and the switch true position should be md-orange-900 not md-orange-700
> * Dark theme:
>   
>   * Jupyter, notebook, json and console icons are not proper. There is a shift to lighter color for the brand and the warning scales. So could you also correct that please.
> * We forget to add fallback value for the switch color CSS properties `--jp-switch-color`, `--jp-switch-true-position-color`

Ok. I will try to correct this in a separate PR

**Review by fcollonval — APPROVED** [review]

**Comment by HaudinFlorence** [comment]
> > Thanks @HaudinFlorence could you please remove some of the dead code.
> > About the notebook icon color, indeed #13279 introduced some regression. Could you correct them in a separate PR.
> > What I saw:
> > 
> > * Light theme:
> >   
> >   * The notebook icon and the switch true position should be md-orange-900 not md-orange-700
> > * Dark theme:
> >   
> >   * Jupyter, notebook, json and console icons are not proper. There is a shift to lighter color for the brand and the warning scales. So could you also correct that please.
> > * We forget to add fallback value for the switch color CSS properties `--jp-switch-color`, `--jp-switch-true-position-color`
> 
> Ok. I will try to correct this in a separate PR

This is done in PR #13500

**Comment by HaudinFlorence** [comment]
Are there other changes that may be useful on this PR ?

**Pull request merged**